src/server/system_services/bucket_server.js

```diff
@@ -1142,24 +1143,30 @@ function get_cloud_buckets(req) {
                 return storage.getBuckets()
                     .then(data => data[0].map(bucket =>
                         _inject_usage_to_cloud_bucket(bucket.name, connection.endpoint, used_cloud_buckets)));
-            } else { //else if AWS
-                var s3 = new AWS.S3({
-                    endpoint: connection.endpoint,
-                    accessKeyId: connection.access_key.unwrap(),
-                    secretAccessKey: connection.secret_key.unwrap(),
-                    signatureVersion: cloud_utils.get_s3_endpoint_signature_ver(connection.endpoint, connection.auth_method),
-                    s3DisableBodySigning: cloud_utils.disable_s3_compatible_bodysigning(connection.endpoint),
-                    httpOptions: {
-                        agent: http_utils.get_unsecured_agent(connection.endpoint)
-                    }
-                });
-                const used_cloud_buckets = cloud_utils.get_used_cloud_targets(['AWS', 'S3_COMPATIBLE', 'FLASHBLADE', 'IBM_COS'],
+            } else if (connection.endpoint_type === 'AWSSTS' || connection.endpoint_type === 'AWS') {
+                if (connection.aws_sts_arn) {
+                    const creds = await cloud_utils.generate_aws_sts_creds(connection, "get_cloud_buckets_session");
+                    connection.access_key = new SensitiveString(creds.accessKeyId);
```

Comment:
don't really like creating this SensitiveString just for unwrapping it later.... just create 2 variables and set them with the right values for both cases (if needed)

Reply:
Done